internal/poller/gotimestatus/gotimestatus.go

```diff
@@ -0,0 +1,188 @@
+
+	lastStatus string
+}
+
+// New constructs a *GoTimeStatus.
+//
+// statusMaxAge sets the max age of a status to notify on
+func New(s Store, c *http.Client, logger zerolog.Logger, statusMaxAge time.Duration, notify NotifyFunc) (*GoTimeStatus, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	lastStatus, notFound, err := s.Get(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get last notitied timestamp: %w", err)
+	}
+
+	if notFound {
+		// doing this explicitly to make sure we are good
+		lastStatus = ""
+
+		if err = s.Put(ctx, lastStatus); err != nil {
+			return nil, fmt.Errorf("failed to initialize redis: %w", err)
+		}
+	}
+
+	return &GoTimeStatus{
+		logger:       logger,
+		store:        s,
+		http:         c,
+		notify:       notify,
+		statusMaxAge: statusMaxAge,
+		lastStatus:   lastStatus,
+	}, nil
+}
+
+type mastodonStatus struct {
+	ID        string     `json:"id"`
+	URL       string     `json:"url"`
+	CreatedAt createTime `json:"created_at"`
+}
+
+type createTime struct {
+	time.Time
+}
+
+func (ct *createTime) UnmarshalJSON(bs []byte) error {
+	var str string
+	if err := json.Unmarshal(bs, &str); err != nil {
+		return err
+	}
+	t, err := time.Parse("2006-01-02T15:04:05.999Z", str)
+	if err != nil {
+		return nil
+	}
+	ct.Time = t
+	return nil
+}
+
+const (
+	// @gotime@changelog.social
+	// acquired from curl 'https://changelog.social/api/v1/accounts/lookup?acct=gotime'
+	// It's immutable so we don't need to look it up again
+	gotimeAccountID = "109349735213354404"
+
+	gotimeStatusesAPI = "https://changelog.social/api/v1/accounts/" + gotimeAccountID + "/statuses"
+)
+
+// Poll conditionally calls notify if there is a status update from @gotime@changelog.social
+//
+// For a status to be posted, it needs to be younger than the maxStatusAge
+// this prevents very old statuses from being notified and acts as a safeguard
+// if the last status ID could not be persisted to state storage, and prevents reposts if we lose the last status ID.
+func (gt *GoTimeStatus) Poll(ctx context.Context) error {
+	gt.logger.Trace().Msg("gotime status poll")
+	now := timeNow()
+	accountStatusURL := gotimeStatusesAPI
+	if gt.lastStatus == "" {
+		gt.logger.Trace().Msg("getting latest status")
+	} else {
+		gt.logger.Trace().Msgf("getting statuses since %s", gt.lastStatus)
+		accountStatusURL = fmt.Sprintf("%s?since_id=%s", gotimeStatusesAPI, url.QueryEscape(gt.lastStatus))
+	}
+	var statuses []mastodonStatus
+	err := gt.get(ctx, accountStatusURL, &statuses)
+	if err != nil {
+		return err
+	}
+	if len(statuses) == 0 {
+		// No new statuses
+		gt.logger.Trace().Msg("no statuses found")
+		return nil
+	}
+	if gt.lastStatus == "" {
+		// no last status, only notify on the latest status
+		// which should be the first element in the list
+		statuses = statuses[0:1]
+	} else {
+		// Sort statuses by age in ascending order
+		sort.Slice(statuses, func(i, j int) bool {
```

Comment:
Let's always sort the list descending by `CreatedAt` so we can then always reference the first element in the sorted list, regardless of `lastStatus` state.

Reply:
Done. It's a bit more awkward to loop over now though.